source/libs/utils.ts

```diff
@@ -138,3 +138,26 @@ export function reportBug(featureName: string, bugName: string): void {
 	console.log('Find existing issues:\n' + String(issuesUrl));
 	console.log('Open new issue:\n' + String(newIssueUrl));
 }
+
+export function timeAgo(since: Date): {interval: number; unit: string} {
+	const units = ['second', 'minute', 'hour', 'day', 'year'];
+	const timespans = [1000, 60000, 3600000, 86400000, 31540000000];
+	const ago = since.getTime() - Date.now();
+	const diff = Math.abs(ago);
+
+	let timespan = timespans[0];
+	let unit = units[0];
+	for (let i = 1; i < timespans.length; i++) {
+		if (diff > timespans[i]) {
+			timespan = timespans[i];
+			unit = units[i];
+		}
+	}
+
+	const interval = Math.round(diff / timespan);
+	if (interval > 1) {
+		unit += 's';
+	}
+
+	return {interval, unit};
+}
```

Comment:
How about this?

```js
function timeAgo (date: Date): [number | string, string?] {
    const today = new Date();
    let difference = (today.getTime() - date.getTime()) / 1000 / 3600 / 24; // In days
    let unit = 'day';
    if (difference > 360) {
        difference /= 365;
        unit = 'year';
    } else if (difference > 30) {
        difference /= 30;
        unit = 'month';
    } else if (date.getDate() === today.getDate()) {
        return ['today'];
    } else if (date.getDate() === today.getDate() - 1) {
        return ['yesterday'];
    }
    
    return [
        Math.floor(difference),
        difference < 2 ? unit : unit + 's'
    ];
}
```

Test the output here: https://runkit.com/embed/b7khixb3zrzu

Almost the same function length but accounts for today/yesterday/months and drops seconds/minutes/hours

Reply:
My original timeAgo utility accounted for those descriptions of the time like 'just now/yesterday/last year', because I wanted to be consistent with how Github handled it. I ended up going back to it always being ```[interval] [unit] old``` after playing around with it because I thought it looked more like a metric in the bar that changes and less like the License.

It's another minor point though and the extension styling is up to you, so let me know

<img width="540" alt="Screen Shot 2019-11-22 at 12 28 23 AM" src="https://user-images.githubusercontent.com/3848317/69385323-3c68a800-0cbf-11ea-8b1d-c2033a363f58.png">
<img width="540" alt="Screen Shot 2019-11-22 at 12 29 29 AM" src="https://user-images.githubusercontent.com/3848317/69385324-3c68a800-0cbf-11ea-91bf-903d5bbc34db.png">
<img width="540" alt="Screen Shot 2019-11-22 at 12 33 24 AM" src="https://user-images.githubusercontent.com/3848317/69385473-bef16780-0cbf-11ea-8064-9c2b4676f3b8.png">

Reply:
Why not just the native browser API? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RelativeTimeFormat

Reply:
https://github.com/sindresorhus/refined-github/pull/2560#discussion_r349017077

Reply:
> [#2560 (comment)](https://github.com/sindresorhus/refined-github/pull/2560#discussion_r349017077)

That link points to the icon thread, here's the one about ```Intl.RelativeTimeFormat```.

https://github.com/sindresorhus/refined-github/pull/2560#discussion_r349205427

Reply:
If our intent is to save some code, I think this is the shortest we can do:

```tsx
// https://github.com/github/time-elements#time-ago
function timeAgo (date: Date): string {
	return (<time-ago datetime={date.toISOString()}/>).textContent;
}

// Replace 'ago' with 'old' if present
// Display time in bold
const [time, ago] = timeAgo(date).split(/(?=ago)/);
<span><b>{time}<b> {ago ?? 'old'}<span>
```

Reply:
I played around with it a little more and I think forcing the output to always be ```interval``` ```unit``` **```old```** is pretty key to preventing confusion about what the metric is telling you. Another option I see is to prepend a word like **created** for ```<time-ago>``` returns now/today/last year...

Alternatively here's an implementation that will always force the ```interval``` ```unit``` **```old```** format using the [micro](https://github.com/github/time-elements#micro-format) return (1m, 1h, 1d, 1y), though it's maybe a little complicated.

```typescript
function timeAgo(date: Date): {interval: number; timespan: string} {
	const units = ['minute', 'hour', 'day', 'year']

	const ago = (<time-ago datetime={date.toISOString()} format="micro"/>).textContent; // 1m, 1h, 1d, 1y.
	const [interval, short] = ago!.match(/[a-z]+|[^a-z]+/gi)! as [number, string];

	let timespan = units.find(unit => unit.slice(0, 1) == short)!;
	timespan = interval > 1 ? timespan + 's' : timespan

	return {interval, timespan};
}
```

Really didn't expect there to be so many ways to skin this cat, gee. :see_no_evil:

Reply:
Not sure I like the short time and why are you adding s to the short time? This turns 2m into 2ms (milliseconds)

Reply:
Doesn’t `time-ago` **always** include `ago` already?? From the docs it appears that it only doesn’t when it’s `just now`, which I don’t really care for since nobody will ever see it.

Reply:
I though that was the case from the docs, but looking at how it's [actually implemented](https://github.com/github/time-elements/blob/master/src/relative-time.js#L207) it can return things like last week/today/yesterday. It's also important to handle the ```just now``` possibility since you run into this right after you fork a repo. Side note, the docs say it'll return ```just now```, but it only returns ```now```.